### PR TITLE
revert altered slider-click behaviour

### LINKED ISSUE #11462: revert altered slider click behaviour (#10499)
The current slider behaviour in darktable 3.8.1 is that _wherever_ you click (and optionally drag) on a slider (including its name and current value labels), that click is used to change the slider's value (the triangle on the slider moves to the same horizontal position as your mouse and continues to move when you drag it). Similarly, double-click anywhere resets the value.

This behaviour was changed in #10499 so that clicking on the label/upper-left-hand-side of the slider grabs focus without changing anything and clicking on the value/upper-right-hand-side pops up the edit box. Unfortunately this means that any "value-change" clicks/drags need to be much more precise - the clickable area has roughly halved (I think).

I've used this for a while now and I'm afraid I just can't get used to it - I keep missing the clickable area. Because my habit is to click the slider and only move my mouse once I've moved my eyes back to the image, I often drag without actually affecting the value and it's annoying.

IMO the new functionality does not add anything useful. It's already possible to bring up the edit box with right-click and focusing could equally be done with Ctrl+click without impacting the more useful behaviour. The same (Ctrl+click) could also be applied to parametric mask sliders.

This has already been discussed in the original PR and under a separate issue but I'm raising this with a more complete description for visibility.

### PR DESCRIPTION
**Pull request by elstoc**
Restore the ability to click anywhere in the slider (including the labels) to alter its value.

Resolves #11462

### CODE REVIEW

**Comment by TurboGit** [comment]
I'm not convinced anymore by this. I was for a revert as I had trouble clicking/selecting the slider... But after a month or so of using dt with this I have adjusted by habits and the smaller sensitive area is not a problem anymore. And yes I know that in #11462 I agreed with the revert.

**Comment by elstoc** [comment]
If you look at the comments in the original PR and the issue there are quite a few people who are for a revert of this. I did wait to see how much support it had before I submitted a PR. I still find myself constantly missing the click area on sliders.

**Comment by Jiyone** [comment]
I must admit sometimes it’s like I need to double clic on the slider :/

**Comment by elstoc** [comment]
The smaller selectable area is only worth the pain if the added functionality (click header to focus, click number for popup editor) is particularly useful. Is it? The former could be done differently, the latter already can be done with right-click.

**Comment by jade-nl** [comment]
If I have to choose between the current small, slider only area in master and the wider area one in stable I would choose the current one. Neither are perfect, though.

One thing I really, really do not like about the wider option is that it extends to above the label (in between the label and the slider above it) and only a very tiny amount under the slider. The eligible  area above the slider is way(!) to large and can lead to annoying situations.

I do agree that the current area is rather small, my preferred solution would be that there should be an obvious dead-zone between sliders and that the eligible area around a slider is evenly spaced all around (probably only above and below) and not too wide. Say an area as wide as slider to bottom of the label above it, on both sides.

Not sure if that is easy to implement, so I'll choose for the current version in master if it is a _one or the other_ choice.

Just my 2c.

**Comment by jade-nl** [comment]
@elstoc 

> So how about if ...

That would work for me (depending on how much *slightly reduce* is going to be), but introducing a larger dead area would be good no matter what. I already mentioned what I think to be a reasonable ballpark and somewhat bigger will still be OK in my book. The dead zone and a more evenly spaced clickable area is what I'm looking for. How much is up to the community and their wishes.

**Comment by elstoc** [comment]
I've pushed a suggestion that keeps the clickable area large but still leaves a reasonable buffer zone that starts towards the top of the slider labels.

**Comment by jade-nl** [comment]
@elstoc : That _suggestion/change_ is a lot better and I think I would prefer this one when comparing to the other 2. Especially the space between clickable areas and the positioning.

This is something that you need to use for a bit before you can say something definitive about.

One thing I did notice that might be seen as a point: When the pointer is in the deadzone one of the 2 labels (the bottom one) is highlighted even though it is not moveable/clickable. Probably nitpicking, it wouldn't bother me for sure, but I do want to mention it just in case.

**Comment by elstoc** [comment]
>  When the pointer is in the deadzone one of the 2 labels (the bottom one) is highlighted even though it is not moveable/clickable.

That's harder to do, since this is controlled via CSS using the widget's "hover" property and the non-clickable area is still within the bounds of the widget.

**Comment by aurelienpierre** [comment]
I'm all for this, and my neighbour is too. 

Ok, I didn't ask him but surely he will prefer not hearing me scream every time I miss that bloody bar.

File: src/bauhaus/bauhaus.c [hunk 1]
```diff
@@ -3066,25 +3066,14 @@ static gboolean dt_bauhaus_slider_button_press(GtkWidget *widget, GdkEventButton
         darktable.bauhaus->mouse_x = ex;
         d->is_dragging = 1;
       }
-      else if(ey > darktable.bauhaus->line_height)
+      else if(ey > darktable.bauhaus->line_height/2)
```
Comment (TurboGit) [inline src/bauhaus/bauhaus.c:3069]: ` / 2.0f`

**Comment by Nilvus** [comment]
I'm all for this. Much better than actual master and previous choice. Good job @elstoc.

About label hovered, @jade-nl, it also help to know that we are on slider widget but that could maybe be better but would need a lot more work on bauhaus code.

**Review by TurboGit — APPROVED** [review]
So given the number of positive vote, let's go for this. Thanks!

**Comment by ahrendsen** [comment]
First off, I want to acknowledge that I'm just jumping in here and I want to thank everyone that's already put a lot of thought into the user interface.

My contribution is as someone that is still in the process of learning darktable is that the slider value should be clickable and bring up the popup editor. I think the wider clickable area for adjusting sliders is good, and should be kept, but the value should be clickable(as requested in point 2 of #10478). I think it would be still intuitive and usable if the clickable area of the slider was reduced just under the value text. I say this without knowledge of how difficult it would be to implement, but I feel really strongly that it will make a big difference in the intuitiveness of the interface.

> The smaller selectable area is only worth the pain if the added functionality (click header to focus, click number for popup editor) is particularly useful. Is it? The former could be done differently (e.g. Ctrl+click), the latter already can be done with right-click.

As @elstoc pointed out, this is already accessible with a right-click, but the issue is that right-clicking on a value to edit is not intuitive to a beginning user. I don't have a source or data to backup this claim, but I don't really think its needed.

I'm going to propose quite a few changes, but I think these changes would significantly improve the process of making fine-scale adjustments. 
1. The slider value should be changed from a label to an editable field.
2. Clicking the field places a cursor in the field at the nearest digit. (Not a huge surprise here, but an alternative is to have the entire field selected on a click).
3. Double clicking the field should no longer reset the value to the default. Double clicking the slider should maintain the reset behavior.
4. With the cursor active in a field, the up/down arrows change the value of the digit immediately to the left of the cursor. This duplicates the functionality of fine tuning features with the scroll wheel, but I think is still worthwhile to implement.

If it's easier to implement, the label doesn't have to be changed to a field. Instead, the label value could be pre-filled into the "entry field" in the popup editor when it is clicked. If this route is chosen, there should be some visual indication to show how far the "value clickable area" extends.

On a related note, I think that you should be able to edit sliders with the arrow keys when the popup editor is activated, and that the up/down arrows should have different functionality than the left/right. It's unrelated enough that I'll submit separate issues about those if I feel strongly enough about them.

**Comment by wpferguson** [comment]
The problem with **intuitive** is that it's in the eye of the beholder.  That word gets used repeatedly in issues and requests to justify changes and it's happened enough that we've finally understood there are no behaviors that are "intuitive" to anyone other than the one using the word.

**Comment by ahrendsen** [comment]
@wpferguson I totally understand the subjectivity of my intuitive claim, but I think it's justified in this case. Perhaps instead of "intuitive" I should have said, "is consistent with the behavior of other interfaces that have been used by the user previously". As you rightly point out, this will be different for everyone based on the programs that they have used. In general I totally agree that there should be a healthy amount of skepticism when people claim something would be intuitive. In this particular case (just talking about left-clicking on a number to change it) I don't think there's going to be much difference in opinion on what's intuitive.

I'd like to amend my earlier suggestions to just be that singular idea, that the value label should be a field, and that left-clicking on the value in the slider interface should allow you to edit it.

**Pull request merged**